bittensor/_axon/__init__.py

```diff
@@ -361,18 +364,39 @@ def __init__(
         self,
         receiver_hotkey: str,
         blacklist: Callable = None,
+        path:str = None
     ):
         r"""Creates a new server interceptor that authenticates incoming messages from passed arguments.
         Args:
             receiver_hotkey(str):
                 the SS58 address of the hotkey which should be targeted by RPCs
             black_list (Function, `optional`):
                 black list function that prevents certain pubkeys from sending messages
+            path (:type:`Optional[str]`, `optional`):
+                Path to save axon data.    
         """
         super().__init__()
+        self.path = f"{path}/nonces_dict.txt" if path != None else None
         self.nonces = {}
         self.blacklist = blacklist
         self.receiver_hotkey = receiver_hotkey
+        self.load()
+        self.upper_bound = int(3.154e+17) # 10 years in nano second
+
+    def load(self):
+        if self.path != None and os.path.exists(self.path):
```

Comment:
as far as i know `is` is more encouraged that `==` or `!=` for this cases

I leave you here some references to that:
- https://stackoverflow.com/questions/14247373/python-none-comparison-should-i-use-is-or
- https://codeql.github.com/codeql-query-help/python/py-test-equals-none/